gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -98,24 +98,23 @@ public void setText (BitmapFont font, CharSequence str, int start, int end, Colo
 		glyphRunPool.freeAll(runs);
 		runs.clear();
 
+		for (int i = 1, n = colorStack.size; i < n; i++)
+			colorPool.free(colorStack.get(i));
+		colorStack.clear();
+
 		BitmapFontData fontData = font.data;
 		if (start == end) { // Empty string.
 			width = 0;
 			height = fontData.capHeight;
 			return;
 		}
 
-		if (truncate != null)
-			wrap = true; // Ensures truncate code runs, doesn't actually cause wrapping.
-		else if (targetWidth <= fontData.spaceXadvance * 3) //
-			wrap = false; // Avoid one line per character, which is very inefficient.
+		boolean wrapOrTruncate = (wrap && targetWidth > fontData.spaceXadvance * 3) // Minimum width for wrapping is 3 * spaceXadvance.
+			|| truncate != null;
```

Comment:
In `setText` there are so many variables which keep chaning and changing that I thought it would be good to use more variables which are final.
I didn't use the final keyword here because it is not used elsewhere in this code. Would it be ok to add more "final"s?

Besides, this reduces the linecount by 2 and makes the code a little easier to read.

Reply:
The old code seems fine to me here. The new comments are maybe not as clear.